packages/app/src/App.tsx

```diff
@@ -84,21 +85,25 @@ function App(props: Props) {
             getFeedbackURL={getFeedbackURL}
           />
           <VisConfigProvider>
-            <ErrorBoundary
-              resetKeys={[selectedPath, isInspecting]}
-              FallbackComponent={ErrorFallback}
-            >
-              <Suspense fallback={<EntityLoader isInspecting={isInspecting} />}>
-                {isInspecting ? (
-                  <MetadataViewer
-                    path={selectedPath}
-                    onSelectPath={onSelectPath}
-                  />
-                ) : (
-                  <Visualizer path={selectedPath} />
-                )}
-              </Suspense>
-            </ErrorBoundary>
+            <DimMappingProvider>
```

Comment:
Putting the new zustand store's provider in `App` to support maintaining dim mapping state when toggling inspect mode.